cmd/mdatagen/metrics_v2.tmpl

```diff
@@ -150,6 +150,19 @@ func (mb *MetricsBuilder) Record{{ $name.Render }}DataPoint(ts pdata.Timestamp
 }
 {{ end }}
 
+// Reset resets metrics builder to its initial state.
+func (mb *MetricsBuilder) Reset(options ...metricBuilderOption) {
+	mb.startTime = pdata.NewTimestampFromTime(time.Now())
+	for _, op := range options {
+		op(mb)
+	}
+}
```

Comment:
Why not replacing the *MetricsBuilder entirely by calling the constructor?

Reply:
We can. It's just couple more things the redis scraper needs to pass around (as part of the scraper struct) that are already part of the builder: startTime and user settings. But I think it's ok for this case. I'll update it

Reply:
May be worth documenting when a "reset" is needed, is it not the case that starttime changed?

Reply:
Yes, changed (or reset) startTime is the only reason I can think of.

Reply:
Updated the doc string. Still not sure if we need this method or ask scraper authors to recreate the builder instead. WDYT?